### PR TITLE
AzureStorage 6.1 compatibility

### PR DESCRIPTION
**Pull request by jason-bragg**
Updated azure storage call patterns to be compatible with azure storage 6.1

The issue stemmed from the fact that the call signature for TableOperation.Retrieve<T> changed between azure storage 5.x and 6.x

< 6.x - public static TableOperation Retrieve<TElement>(string partitionKey, string rowkey) where TElement : ITableEntity
\> 6.x - public static TableOperation Retrieve<TElement>(string partitionKey, string rowkey, List<string> selectColumns = null) where TElement : ITableEntity

To address this, the table query call using Retrieve<T> was replaced with a newer azure storage call pattern using TableQuery.

There are a number of different ways to form such a query, but this one was selected because it demonstrated improved performance characteristics over some of the simpler patterns.  This performance testing was done using azure storage 2.1 (months after it was released), and it's findings have not been reexamined since.

### CODE REVIEW

File: src/TesterInternal/App.config [hunk 1]
```diff
@@ -7,5 +7,11 @@
     <ThrowUnobservedTaskExceptions enabled="false"/>
     <gcServer enabled="true"/>
     <gcConcurrent enabled="false"/>
+    <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
```
Comment (sergeybykov) [inline src/TesterInternal/App.config]: Is this somehow related?

Comment (jason-bragg) [inline src/TesterInternal/App.config]: Loosely.  When testing, I found that some of our current azure tests, even without my code changes, were failing because they were missing this redirect.  Adding this redirect fixed these tests, and allowed me to vet this code change.
